vcr/pe/presentation_submission.go

```diff
@@ -254,3 +255,61 @@ func resolveCredential(path []string, mapping InputDescriptorMappingObject, valu
 	_ = json.Unmarshal(decodedValueJSON, &decodedValueMap)
 	return resolveCredential(fullPath, *mapping.PathNested, decodedValueMap)
 }
+
+// Validate validates the Presentation Submission to the Verifiable Presentations and Presentation Definitions and returns the mapped credentials.
+// The credentials will be returned as map with the InputDescriptor.Id as key.
+// The Presentation Definitions are passed in the envelope, as specified by the PEX specification.
+// It assumes credentials of the presentations only map in 1 way to the input descriptors.
+func (s PresentationSubmission) Validate(envelope interface{}, definition PresentationDefinition) (map[string]vc.VerifiableCredential, error) {
+	actualCredentials, err := s.Resolve(envelope)
+	if err != nil {
+		return nil, fmt.Errorf("resolve credentials from presentation submission: %w", err)
+	}
+
+	// Create a new presentation submission the being validated should map the Input Descriptors to the same credentials.
+	// First, create a new submission
+	var presentations []vc.VerifiablePresentation
+	envelopeJSON, _ := json.Marshal(envelope)
+	if _, ok := envelope.([]interface{}); ok {
+		if err := json.Unmarshal(envelopeJSON, &presentations); err != nil {
+			return nil, fmt.Errorf("unable to unmarshal envelope: %w", err)
+		}
+	} else {
+		var presentation vc.VerifiablePresentation
+		if err := json.Unmarshal(envelopeJSON, &presentation); err != nil {
+			return nil, fmt.Errorf("unable to unmarshal envelope: %w", err)
+		}
+		presentations = []vc.VerifiablePresentation{presentation}
+	}
+	submissionBuilder := definition.PresentationSubmissionBuilder()
```

Comment:
if the envelope is sued at more places in the future, it could get it's own type?

Reply:
Could, but it would just be an alias for `interface{}`, so not very useful?